### PR TITLE
Ignore approvals from robot users

### PR DESCRIPTION
**Pull request by jmcs**
Currently robot users can create approval comments as part of the build process and they will be counted.

### CODE REVIEW

File: server/checks/Approval.js [hunk 1]
```diff
@@ -467,6 +467,11 @@ export default class Approval extends Check {
           debug(`${repository.full_name}#${issue.number}: Ignoring comment, not a PR`)
           return
         }
+        const author = hookPayload.comment.user.login
+        if (author.endsWith('-robot')) {
```
Comment (hjacobs) [inline server/checks/Approval.js]: This looks like a hack, shouldn't this be configurable?

Comment (harti2006) [reply server/checks/Approval.js]: Yeah, or maybe a list of suffixes or regexes would be great.

**Comment by harti2006** [comment]
:+1:

**Comment by jmcs** [comment]
@harti2006 @hjacobs I removed the configuration since it was breaking UI tests. I would hardcode it for now and create an issue to make it configurable.

**Comment by jmcs** [comment]
:+1:

File: server/checks/Approval.js [hunk 2]
```diff
@@ -6,11 +6,24 @@ import * as EVENTS from '../model/GithubEvents'
 import * as AUDIT_EVENTS from '../service/audit/AuditEventTypes'
 import * as _ from 'lodash'
 
+const IGNORE_LOGIN_RE =[/-robot^/]
```
Comment (harti2006) [inline server/checks/Approval.js]: I just realized, that the regex should be `/-robot$/`. Can you maybe add a small unit test?

**Comment by harti2006** [comment]
:+1:

**Comment by jmcs** [comment]
:+1:

**Comment by rashamalek** [comment]
deployed on https://zappr.opensource.zalan.do

**Pull request merged**